### PR TITLE
Fix exception in case of compressing empty logs

### PR DESCRIPTION
**Pull request by pmisik**
## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the master/buildbot/newsfragment directory (and read the README.txt in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: master/buildbot/test/unit/test_db_logs.py [hunk 1]
```diff
@@ -517,6 +517,11 @@ def test_addLogLines_huge_log_lots_line(self):
     def test_addLogLines_huge_log_lots_snowmans(self):
         return self.do_addLogLines_huge_log(NUM_CHUNKS=3000, chunk=u'\N{SNOWMAN}\n' * 50)
 
+    @defer.inlineCallbacks
+    def test_compressEmptyLog(self):
+        yield self.db.logs.compressLog(201)
+        logdict = yield self.db.logs.getLog(201)
+        self.assertEqual(logdict, None)
```
Comment (seankelly) [inline master/buildbot/test/unit/test_db_logs.py]: Need to add one more blank line after the new method so there are two blank lines between the method and class.

File: master/buildbot/test/unit/test_db_logs.py [hunk 2]
```diff
@@ -517,6 +517,12 @@ def test_addLogLines_huge_log_lots_line(self):
     def test_addLogLines_huge_log_lots_snowmans(self):
         return self.do_addLogLines_huge_log(NUM_CHUNKS=3000, chunk=u'\N{SNOWMAN}\n' * 50)
 
+    @defer.inlineCallbacks
+    def test_compressEmptyLog(self):
```
Comment (tardyp) [inline master/buildbot/test/unit/test_db_logs.py]: This test tests the not existing log use case (which is nice to test too, but not coherent with the test name)

In order to test the EmptyLog use case, you need to insert the following in the db:

```python
        yield self.insertTestData(self.backgroundData + [
            fakedb.Log(id=201, stepid=101, name=u'stdio', slug=u'stdio',
                       complete=1, num_lines=0, type=u's'),
        ])
```

**Comment by pmisik** [comment]
Thank you for review. 
Original test is renamed and added requested one.

**Comment by tardyp** [comment]
great! @pmisik congrats for your first github contribution!

**Pull request merged**